app/controllers/automated_tests_controller.rb

```diff
@@ -78,7 +78,16 @@ def student_interface
 
     unless @grouping.nil?
       @grouping.refresh_test_tokens
+      # authorization
+      begin
+        authorize! @grouping, to: :run_tests?
+        @authorized = true
+      rescue ActionPolicy::Unauthorized => e
+        @authorized = false
+        @reason = e.result.reasons.full_messages.join(' ')
```

Comment:
I'd prefer keeping flash messages in the controller rather than the view.

Reply:
This change made me discover a regression in the messages displayed, a bit more work to do :(